src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -71,6 +71,159 @@ Pane::Pane(const GUID& profile, const TermControl& control, const bool lastFocus
     });
 }
 
+// Method Description:
+// - Extract the terminal settings from the current (leaf) pane's control
+//   to be used to create an equivalent control
+// Arguments:
+// - <none>
+// Return Value:
+// - Arguments appropriate for a SplitPane or NewTab action
+NewTerminalArgs Pane::GetTerminalArgsForPane() const
+{
+    // Leaves are the only things that have controls
+    assert(_IsLeaf());
+
+    NewTerminalArgs args{};
+    auto controlSettings = _control.Settings().as<TerminalSettings>();
+
+    args.Profile(controlSettings.ProfileName());
+    args.StartingDirectory(controlSettings.StartingDirectory());
+    args.TabTitle(controlSettings.StartingTitle());
+    args.Commandline(controlSettings.Commandline());
+    args.SuppressApplicationTitle(controlSettings.SuppressApplicationTitle());
+    if (controlSettings.TabColor() || controlSettings.StartingTabColor())
+    {
+        til::color c;
+        // StartingTabColor is prioritized over other colors
+        if (controlSettings.StartingTabColor())

```

Comment:
BTW In such cases you can write:
```cpp
if (const auto color = controlSettings.StartingTabColor())
```
...and use the non-null variable directly.

Reply:
I tried doing this, but I just get that `color` is also an `IReference<...>` like `StartingTabColor()` instead of unwrapping it.

Reply:
I meant like this:
```cpp
til::color c;
if (const auto color = controlSettings.StartingTabColor())
{
    c = til::color(color.Value());
}
else
{
    c = til::color(controlSettings.TabColor().Value());
}
```

Reply:
Feels like a marginal improvement, but I can make it.